processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -556,9 +564,37 @@ public byte[] computeResultLevelCacheKey(GroupByQuery query)
             builder.appendStrings(subTotalSpec);
           }
         }
+        if (query.getLimitSpec() instanceof DefaultLimitSpec) {
```

Comment:
This is dealing with the orderbys, but I think there's still an issue with postaggregators in groupBy result-level cache keys: they refer to aggregators and dimensions (and other post-aggregators) by name, but the names aren't stored in the cache key.

I think we can deal with that by storing `List<String>`s corresponding to the names of all dims, aggs, and postaggs. That's equivalent to adding the name to each of those items' cache keys.

Reply:
dimension names are stored in their cache key, but aggs are not (consistently at least, though some are). Yeah, that would resolve the result level cache issue, though it is a shame that cosmetic stuff could reduce the cache hit rate when all that _really_ matters is that everything is in the same place and ordered/limited the same.

I wonder if instead of anything using name we should rework this sometime in the future to use position or something...